[BUG] btp_subaccount_role (Resource)

### Is there an existing issue for this?

- [x] I have searched the existing issues

### What version of the Terraform provider are you using?

1.17.0

### What version of the Terraform CLI are you using?

1.12.2

### What type of issue are you facing

bug report

### Describe the bug

Error: Provider produced inconsistent result after apply

When applying changes to
module.iac-module-btp-subaccount_role["cmp-tas-shd-ap29839-pro-jkey,cmp-RP-ImportSelectedOperator-cmp-tas-shd-ap29616-pro,ImportSelectedOperator,alm-ts!t98"].btp_subaccount_role.btp_subaccount_role,
provider "provider[\"registry.terraform.io/sap/btp\"]" produced an unexpected
new value: .attribute_list[0].value_required: was cty.True, but now
cty.False.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.
Releasing state lock. This may take a few moments...
Error: Terraform exited with code 1.
Error: Process completed with exit code 1.

### Expected Behavior

The new role is created once the call to our module is done.
The variable declaration on our module is like;

variable "subaccountsRoles" {
  type = object({
    subaccount_id      = string
    name               = string
    role_template_name = string
    app_id             = string
    description        = optional(string)
    attribute_list = optional(list(object({
      attribute_name         = string
      attribute_value_origin = string
      attribute_values       = list(string)   // Changed to list of strings to accommodate multiple values
      value_required         = optional(bool) // Added to reflect the optional requirement
    })))
  })
}


### Steps To Reproduce

terraform -chdir=./ apply -target module.iac-module-btp-subaccount_role_collection [...] 
-var "globalaccount=globalAccounbtID"  
"-var-file=terraform.globalAccountRolesAndUsers.json" [...] 
-no-color -input=false -auto-approve

### User's Role Collections

 ```terraform
 # module.iac-module-btp-subaccount_role["cmp-tas-shd-ap29839-pro-jkey,cmp-RP-ImportSelectedOperator-cmp-tas-shd-ap29616-pro,ImportSelectedOperator,alm-ts!t98"].btp_subaccount_role.btp_subaccount_role will be created
  + resource "btp_subaccount_role" "btp_subaccount_role" {
      + app_id             = "alm-ts!t98"
      + attribute_list     = [
          + {
              + attribute_name         = "TmsNodesImportSelected"
              + attribute_value_origin = "static"
              + attribute_values       = [
                  + "cmp-tas-shd-ap29616-pro",
                ]
              + value_required         = true
            },
        ]
      + description        = (known after apply)
      + id                 = (known after apply)
      + name               = "cmp-RP-ImportSelectedOperator-cmp-tas-shd-ap29616-pro"
      + read_only          = (known after apply)
      + role_template_name = "ImportSelectedOperator"
      + subaccount_id      = "cmp-cmp-cmp-cmp-cmp8e"
    }

  # module.iac-module-btp-subaccount_role["cmp-tas-shd-ap29839-pro-jkey,cmp-RP-ImportSelectedOperator-cmp-tas-shd-ap29616-qua,ImportSelectedOperator,alm-ts!t98"].btp_subaccount_role.btp_subaccount_role will be created
  + resource "btp_subaccount_role" "btp_subaccount_role" {
      + app_id             = "alm-ts!t98"
      + attribute_list     = [
          + {
              + attribute_name         = "TmsNodesImportSelected"
              + attribute_value_origin = "static"
              + attribute_values       = [
                  + "cmp-tas-shd-ap29616-qua",
                ]
              + value_required         = true
            },
        ]
      + description        = (known after apply)
      + id                 = (known after apply)
      + name               = "cmp-RP-ImportSelectedOperator-cmp-tas-shd-ap29616-qua"
      + read_only          = (known after apply)
      + role_template_name = "ImportSelectedOperator"
      + subaccount_id      = "cmp-cmp-cmp-cmp-cmp8e"
    }
```

### Add screenshots to help explain your problem

<img width="1402" height="219" alt="Image" src="https://github.com/user-attachments/assets/135f08a8-c92b-41a5-8ba1-1370f226a8c1" />

### Additional context

What is the value expected for this 
      value_required         = optional(bool) // Added to reflect the optional requirement

How we can determinate how this value_required is populated?

--- a/btp/provider/resource_subaccount_role.go
+++ b/btp/provider/resource_subaccount_role.go
@@ -120,7 +120,8 @@ __Further documentation:__
 							},
 						},
 						"value_required": schema.BoolAttribute{
-							MarkdownDescription: "Shows whether the value is required.",
+							MarkdownDescription: "Do not use this attribute as it is defined is defined in the security descriptor (see <https://help.sap.com/docs/btp/sap-business-technology-platform/application-security-descriptor-configuration-syntax?#loio517895a9612241259d6941dbf9ad81cb__section_pv5_5qr_xs>).",
+							DeprecationMessage:  "Remove this attribute's configuration as it is not relevant for role creation. The attribute is defined in the security descriptor (see <https://help.sap.com/docs/btp/sap-business-technology-platform/application-security-descriptor-configuration-syntax?#loio517895a9612241259d6941dbf9ad81cb__section_pv5_5qr_xs>) and not relevant for role creation.",
 							Optional:            true,
 							Computed:            true,
 						},

--- a/docs/resources/subaccount_role.md
+++ b/docs/resources/subaccount_role.md
@@ -76,7 +76,7 @@ Required:
 
 Optional:
 
-- `value_required` (Boolean) Shows whether the value is required.
+- `value_required` (Boolean, Deprecated) Do not use this attribute as it is defined is defined in the security descriptor (see <https://help.sap.com/docs/btp/sap-business-technology-platform/application-security-descriptor-configuration-syntax?#loio517895a9612241259d6941dbf9ad81cb__section_pv5_5qr_xs>).
 
 ## Import
 

